modules/core/02-client/keeper/keeper.go

```diff
@@ -31,10 +34,11 @@ type Keeper struct {
 	paramSpace    paramtypes.Subspace
 	stakingKeeper types.StakingKeeper
 	upgradeKeeper types.UpgradeKeeper
+	authority     string
 }
 
 // NewKeeper creates a new NewKeeper instance
-func NewKeeper(cdc codec.BinaryCodec, key storetypes.StoreKey, paramSpace paramtypes.Subspace, sk types.StakingKeeper, uk types.UpgradeKeeper) Keeper {
+func NewKeeper(cdc codec.BinaryCodec, key storetypes.StoreKey, paramSpace paramtypes.Subspace, sk types.StakingKeeper, uk types.UpgradeKeeper, authority string) Keeper {
```

Comment:
If we're going to go ahead with this approach we should cherry pick this API breaking change out to a separate PR that we can backport to v7

Reply:
do you just mean the method signature change? Not the `MsgUpdateParams` ?

Reply:
Yeah, just the method signature change and addition of the `authority` field on the keeper struct